sklearn/preprocessing/tests/test_data.py

```diff
@@ -1202,6 +1207,7 @@ def test_normalizer_l2():
         X = init(X_dense)
         X_norm = normalizer = Normalizer(norm='l2', copy=False).transform(X)
 
+        assert_equal(X_norm.dtype, X.dtype)
```

Comment:
If I change X to `npfloat.32` here, it will result in 
```
AssertionError:
Arrays are not almost equal to 7 decimals
 ACTUAL: 0.99999994
 DESIRED: 1.0
```
caused by this [line](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/preprocessing/tests/test_data.py#L1210). 

But it will pass 
```python
assert_equal(X_norm.dtype, X.dtype) 
```

Maybe we can leave dtype test [here](https://github.com/scikit-learn/scikit-learn/pull/6539/files#diff-274f1031827d26cb7a055b88042b45f1R494) which checks dtype of X is preserved?